opslevel/resource_opslevel_repository.go

```diff
@@ -1,99 +1,264 @@
-
-// 	return nil
-// }
-
-// func resourceRepositoryUpdate(d *schema.ResourceData, client *opslevel.Client) error {
-// 	input := opslevel.RepositoryUpdateInput{
-// 		Id: *opslevel.NewID(d.Id()),
-// 	}
-
-// 	if owner, ok := d.GetOk("owner"); ok {
-// 		input.OwnerId = opslevel.NewID(owner.(string))
-// 	} else {
-// 		input.OwnerId = opslevel.NewID("")
-// 	}
-
-// 	_, err := client.UpdateRepository(input)
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	d.Set("last_updated", timeLastUpdated())
-// 	return resourceRepositoryRead(d, client)
-// }
-
-// func resourceRepositoryDelete(d *schema.ResourceData, client *opslevel.Client) error {
-// 	// No API call to make because the repository is not able to be deleted
-// 	d.SetId("")
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &RepositoryResource{}
+
+var _ resource.ResourceWithImportState = &RepositoryResource{}
+
+func NewRepositoryResource() resource.Resource {
+	return &RepositoryResource{}
+}
+
+// RepositoryResource defines the resource implementation.
+type RepositoryResource struct {
+	CommonResourceClient
+}
+
+// RepositoryResourceModel describes the Repository managed resource.
+type RepositoryResourceModel struct {
+	Id          types.String `tfsdk:"id"`
+	Identifier  types.String `tfsdk:"identifier"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+	Owner       types.String `tfsdk:"owner"`
+}
+
+func NewRepositoryResourceModel(ctx context.Context, repository opslevel.Repository) RepositoryResourceModel {
+	return RepositoryResourceModel{Id: ComputedStringValue(string(repository.Id))}
+}
+
+func (r *RepositoryResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_repository"
+}
+
+func (r *RepositoryResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Repository Resource",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the repository.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"identifier": schema.StringAttribute{
+				Description: "The id or human-friendly, unique identifier for the repository.",
```

Comment:
Overall I find this field (and the old version too) pretty confusing - shouldn't this be called `alias`? I don't think that an ID can be passed here. 🤔

Reply:
ID is used in one spot down below to get the repo in the create operation.
	`repository, err = r.client.GetRepository(*opslevel.NewID(identifier))`